wagtail/wagtailcore/models.py

```diff
@@ -391,6 +391,15 @@ def _get_autogenerated_slug(self, base_slug):
         candidate_slug = base_slug
         suffix = 1
         parent_page = self.get_parent()
+        slug_parts = base_slug.split('-')
+
+        try:
+            slug_parts[-1] = str(int(slug_parts[-1]) + 1)
+            candidate_slug = '-'.join(slug_parts)
+        except ValueError:
+            # If the last part of the slug fails to be converted
+            # to an int then simply move on to auto incrementing
+            pass
 
         while not Page._slug_is_available(candidate_slug, parent_page, self):
             # try with incrementing suffix until we find a slug which is available
```

Comment:
I didn't touch the original logic here so I didn't break anything, but by the looks of it the code simply appends -2 to every slug even if it ends in an integer. So if for instance you have 'some-slug-5' and you have a duplicate you end up with 'some-slug-5-2' instead of 'some-slug-6'